site-src/versioning.md

```diff
@@ -0,0 +1,102 @@
+Adapted with :blue_heart: from the [gateway api project release documentation](https://gateway-api.sigs.k8s.io/)
+
+## Overview
+Each new release of NetworkPolicy APIs is defined with a "bundle version" that
+represents the Git tag of a release, such as v0.4.0. This contains the
+following:
+
+* API Types (Go bindings for the resources)
+* CRDs (Kubernetes definitions of the resources)
+
+### API Versions (e.g. v1alpha2, v1beta1)
+Within Network Policy API, API versions are primarily used to indicate the stability of a resource. For example, if a resource has not yet graduated to beta, it is still possible that it could either be removed from the API or changed in
+backwards incompatible ways. For more information on API versions, refer to the
+[full Kubernetes API versioning
+documentation](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+## Version Indicators
+Each CRD will be published with annotations that indicate their bundle version
+and channel:
+
+```
+policy.networking.k8s.io/bundle-version: v0.4.0
+```
+
+## What can Change
+When using or implementing this API, it is important to understand what can
+change across API versions.
+
+### Patch version (e.g. v0.4.0 -> v0.4.1)
+* Clarifications to godocs
+* Updates to CRDs and/or code to fix a bug
+* Conformance test fixes
+* Additional conformance test coverage for existing features
+* Fixes to typos
+
+### Minor version (e.g. v0.4.0 -> v0.5.0)
+* Everything that is valid in a patch release
+* New API fields or resources
+* Changes to recommended conditions or reasons in status
+* Loosened validation
+* Making required fields optional
+* Removal of fields
+* Removal of resources
+* Changes to conformance tests to match spec updates
+* Introduction of a new **API version**, which may also include:
+  * Renamed fields
+  * Anything else that is valid in a new Kubernetes API version
+  * Removal/tombstoning of beta fields
+* Removal of an API resource following [Kubernetes deprecation
+  policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+### Major version (e.g. v0.x to v1.0)
+* There are no API compatibility guarantees when the major version changes.
+
+## Graduation Criteria
+
+### Resources
+
+#### Alpha -> Beta
+A resource to graduate from alpha to beta must meet the following criteria:
+
+* Implemented by two implementations.
+* Conformance test framework is in place, with some coverage of basic
+  functionality.
+* Validation is well thought out.
+* Most of the API surface is being exercised by users.
+* Approval from subproject owners + KEP reviewers.
+
+#### Beta -> GA
+
+A resource to graduate from beta to GA must meet the following criteria:
+
+* Almost all of the fields and behavior have conformance test coverage.
+* Multiple conformant implementations.
+* Widespread implementation and usage.
+* At least 6 months of soak time as a beta API.
+* Approval from subproject owners + KEP reviewers.
```

Comment:
We can add scale testing here.

Reply:
Hrm Idk how to frame this do we want "upstream scale testing?" It's hard to say "All features are proven to be scalable"

Reply:
Ah hmm something like "ensuring API fields are scalable" is good, surely burden is on implementations, BUT I would like to reconsider things like supporting upto 1000 priorities and same/nodeSame etc if no one can use it due to it not scaling well across multiple implementations...